source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -889,10 +889,8 @@ void ServerContextImpl::populateServerNamesMap(TlsContext& ctx, int pkey_id) {
     auto sn_match = server_names_map_.try_emplace(sn_pattern, pkey_types_map).first;
     auto pt_match = sn_match->second.find(pkey_id);
     if (pt_match != sn_match->second.end()) {
-      throw EnvoyException(fmt::format(
-          "Failed to load certificate chain from {}, at most one "
-          "certificate of a given type may be specified for each DNS SAN entry or Subject CN: {}",
-          ctx.cert_chain_file_path_, sn_match->first));
+      // When there are duplicate names, prefer the earlier one.
+      return;
```

Comment:
I think we should at least warn about this? This seems like something the operator should know about?

Reply:
From some conversations I've had, I think it's common enough (and intentional) that I'm not sure it deserves a warning. I think in most cases that will just be noise in the logs.

Reply:
It's probably not the entire cert that won't be used; just one of the SANs.

With some more tracking, I could figure out if any cert is completely shadowed for all SANs by other certs. That seems worth warning about. But it adds complexity, and I don't know that it's needed.

Reply:
Oh I see, OK. Yeah maybe just add a TODO/comment about that.